mapper/cumulocity/c8y_smartrest/src/smartrest_serializer.rs

```diff
@@ -96,13 +118,23 @@ impl<'a> SmartRestSerializer<'a> for SmartRestSetOperationToExecuting {}
 pub struct SmartRestSetOperationToSuccessful {
     pub message_id: &'static str,
     pub operation: &'static str,
+    pub url: Option<String>,
 }
 
 impl SmartRestSetOperationToSuccessful {
     fn new(operation: CumulocitySupportedOperations) -> Self {
         Self {
             message_id: "503",
             operation: operation.into(),
+            url: None,
+        }
+    }
+
+    pub fn new_with_file(operation: CumulocitySupportedOperations, url: &str) -> Self {
+        Self {
+            message_id: "503",
+            operation: operation.into(),
+            url: Some(url.into()),
         }
```

Comment:
Or another way to refactor this (and make it similar to other parts of the codebase) would be to use builder-ish pattern:
```suggestion
    fn new(operation: CumulocitySupportedOperations) -> Self {
        Self {
            message_id: "503",
            operation: operation.into(),
            url: None,
        }
    }

    with_response_parameter(self, response_parameter: &str) -> Self {
        Self {
            response_parameter: Some(url.into()),
            ..self
        }
```

And then can be used as: `SmartRestSetOperationToSuccessful::new("aaa").with_response_parameter("paramterv_value")`.